mindsdb/integrations/handlers/duckdb_handler/duckdb_handler.py

```diff
@@ -125,8 +123,7 @@ def native_query(self, query: str) -> Response:
             response = Response(RESPONSE_TYPE.ERROR, error_message=str(e))
 
         cursor.close()
-        if need_to_close is True:
```

Comment:
Why has this call to `self.disconnect()` been removed?

Reply:
@MinuraPunchihewa Can you check carefully? I havn't remove the call to `self.disconnect()` method, only removed the condition. So that the the `disconnect` method executes every time.

Reply:
@MinuraPunchihewa 
Can you check it carefully? I havn't remove the call to `self.disconnect()` method, only removed the condition. So that the the `disconnect` method executes every time.

Reply:
Because of that condition I was unable to sync the knowledge base. You can see the demo [here](https://github.com/mindsdb/mindsdb/issues/10043#issuecomment-2439634021) .

Reply:
We need to be careful when removing checks that work only for specific use cases.

Reply:
Yes, I am also agree with  @ZoranPandovski, For this type of situation we can check git logs and look whether these lines are changed previously or not for any reason.

In my case that `need_to_close is True:` condition was there from the begining of the handler.

![duck](https://github.com/user-attachments/assets/227a3856-33cb-4382-a904-daa68007b6ad)

Reply:
Removing the if condition shouldn't be the way to solve this, closing this PR